### DISCUSSION TITLE
image (files) upload limit?

### DISCUSSION THREAD

**Discussion by Nico-L** [nhost/nhost#143] [Q&A]
Hello,

I like working with hasura but always had an issue with auth. Nhost seems to be a good way to add this feature and storage so I tried it for my current project.

I usually need a way to upload images at some point of my work. I encoutered some issues using nhost. Some of them has been resolved put I put it here anyway in case someone else has the same question :
* in my projects, the user usually has the possibility to upload images. In order not to send huge files (which is very limited anyway with nhost, see the following point), an image resize is performed before upload. The way I usually do it is to output a base64 string. It is not possible to upload string via the nhost.storage upload. So I output a blob to fix the issue
* the resize of the image asks for a max width/height size. I usually limit to 2048px in order to have multiple use of the image (banner down to thumbnail). I am facing an upload limit by nhost low enough not to be able to resize to 1024px, which is too low for my needs. What is the actual upload file size limit? I didn't find it in the docs. Is it different between free and paid version?
* On a low priority end, it would be interesting to have the list of list present in the different buckets independantly of the files table to have an idea of orphan files? In the nhost dashboard for example. I deleted some entries in the table directly from hasura console and so some files may remain in the bucket. I am not sure whether when queries go wrong it is possible to have a file table entry deleted without the file being actually removed.

Thanks

**Comment by elitan** [comment]
1. It's currently not yet supported to upload a string with our storage solution but it's on our roadmap.
2. You can specify the maximum file size per bucket in the `storage.buckets` table. There is no difference between the free/paid plan regarding size limits for individual files.
3. Yes I agree with having a UI for storage in our dashboard. It's something we have plans for in the future. Same things for removing orphans. We'll implement and document the way this is supposed to work.
